app/Http/Controllers/ScoresController.php

```diff
@@ -54,6 +61,18 @@ public function download($rulesetOrSoloId, $id = null)
             'Content-Type' => 'application/x-osu-replay',
         ];
 
+
+        if (Auth::user()->user_id !== $score->user->user_id) {
+            $score->user->statistics($ruleset, true)->increment('replay_popularity');
+
+            $currentMonth = UserCountryHistory::formatDate(CountryChangeTarget::currentMonth());
+            $score->user->replaysWatchedCounts->where('year_month', $currentMonth)->first()->increment('count');
```

Comment:
this would fetch all rows first before finding the correct one which is rather wasteful (and then explode if the row doesn't exist)

Reply:
isn't fetching all rows unavoidable when accessing it with `$score->user->replaysWatchedCounts`? I guess if I query `UserReplaysWatchedCount` directly then I can avoid it

Reply:
use `replaysWatchedCounts()` like everywhere else?

Reply:
ah, whoops